File: Proyecto DAWA Angular/src/app/models/Producto.ts
import { Categoria } from './Categoria';

export class Producto {
  idProducto: number;
  nombre: string;
  descripcion: string;
  categoria: Categoria;
  precioUnitario: number;
  stock: number;

  constructor(
    _idProducto: number,
    _nombre: string,
    _descripcion: string,
    _categoria: Categoria,
    _precioUnitario: number,
    _stock: number
  ) {
    this.idProducto = _idProducto;
    this.nombre = _nombre;
    this.descripcion = _descripcion;
    this.categoria = _categoria;
    this.precioUnitario = _precioUnitario;
    this.stock = _stock;
  }
}

File: ProyectoBackEnd/Proyecto/Models/Producto.cs
using System;
using System.Collections.Generic;

namespace Proyecto.Models
{
    public partial class Producto
    {
        public Producto()
        {
            DetalleVenta = new HashSet<DetalleVentum>();
        }

        public int IdProducto { get; set; }
        public string? Nombre { get; set; }
        public int? IdCategoria { get; set; }
        public int? Stock { get; set; }
        public decimal? Precio { get; set; }
        public bool? EsActivo { get; set; }
        public DateTime? FechaRegistro { get; set; }

        public virtual Categorium? IdCategoriaNavigation { get; set; }
        public virtual ICollection<DetalleVentum> DetalleVenta { get; set; }
    }
}
